### PR TITLE
[Merged by Bors] - SVM: remove `ValidateAndAddTxToPool`

### LINKED ISSUE #2888: Removal of deprecated `SVM.ValidateAndAddTxToPool`
## Description
The `SVM.ValidateAndAddTxToPool` method has been soft deprecated in https://github.com/spacemeshos/go-spacemesh/pull/2864 and it must be completely removed before SVM integration is complete. See the SVM epic https://github.com/spacemeshos/pm/issues/89 for more information and tracking. We plan on separating mempool management with SVM state manipulation logic.

## Affected code

- `svm/api.go`.
- `svm/state/processor.go`.
- `mesh/mesh.go`.
- Testing, mocking.

### PR DESCRIPTION
**Pull request by neysofu**
## Motivation

Closes https://github.com/spacemeshos/go-spacemesh/issues/2888.

## Changes

- Split transaction validation and mempool state manipulation into separate operations, akin to the "SVM way".
- Getters for `tx.Fee` and `tx.Recipient`, in preparation for direct access hard deprecation.

## Test Plan

No functional changes.

## DevOps Notes
- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by neysofu** [comment]
bors try

**Comment by neysofu** [comment]
bors try

File: common/types/transaction.go [hunk 1]
```diff
@@ -118,6 +118,16 @@ func (t *Transaction) ID() TransactionID {
...
 
+// GetFee returns the fee of the transaction.
+func (t *Transaction) GetFee() uint64 {
+	return t.Fee
+}
+
+// GetRecipient returns the transaction recipient.
+func (t *Transaction) GetRecipient() Address {
+	return t.Recipient
+}
```
Comment (countvonzero) [inline common/types/transaction.go:129]: what's the reason for these getters? seem redundant since Fee and Recipient are meant to be public

**Review by countvonzero — APPROVED** [review]

File: common/types/transaction.go (hunk 1 above)
Comment (neysofu) [reply common/types/transaction.go:129]: This is in preparation for removing the fields and deprecating direct access, further down the road in SVM integration

**Review by noamnelke — APPROVED** [review]

**Comment by neysofu** [comment]
bors merge

**Pull request closed without merging**